### PR TITLE
Allow strings py3

### PR DESCRIPTION
**Pull request by kelvinatorr**
This allows users to pass in strings for keys and values in python3 which makes the interface similar to how it was in python2. I also added a bunch more unittests.

### CODE REVIEW

File: mtbl.pyx [hunk 1]
```diff
@@ -120,7 +148,7 @@ cdef class iterkeys(object):
             res = mtbl_iter_next(self._instance, &key, &len_key, &val, &len_val)
         if res == mtbl_res_failure:
             raise StopIteration
-        return <bytes> key[:len_key]
+        return from_bytes(key[:len_key])
```
Comment (hstern) [inline mtbl.pyx:151]: I feel a bit funny with this being the default behaviour with no way to disable it. How about we add a kwargs bool to __init__ to turn it off?

**Review by hstern — COMMENTED** [review]
Love all of the tests.

File: README.md [hunk 2]
```diff
@@ -9,45 +9,80 @@ writer interfaces (nb. the traceback is expected behavior):
...
+import mtbl
+w = mtbl.writer('example.mtbl', compression=mtbl.COMPRESSION_SNAPPY)
+w['key0'] = mtbl.varint_encode(2)
+w['key1'] = mtbl.varint_encode(128)
+w.close()
+
+r = mtbl.reader('example.mtbl', verify_checksums=True)
+
+assert mtbl.varint_decode(r.get('key0')[0]) == 2
+assert mtbl.varint_decode(r.get('key1')[0]) == 128
```
[the remark this answers is not in the document]
Comment (kelvinatorr) [reply README.md:91]: It actually works. I tried:

assert 987654321000 == mtbl.varint_decode(mtbl.varint_encode(987654321000))

and it worked. I added it as an example so it is clear.

File: README.md [hunk 3]
```diff
@@ -9,45 +9,80 @@ writer interfaces (nb. the traceback is expected behavior):
...
+
+assert r.get(b'key0') == ['val0']
+# all the keys in this mtbl are strings because they can all be UTF-8 decoded
+assert [type(k) for k in list(r.iterkeys())] == [str, str, str, str]
+assert r.get('key1') == ['\x01']
+assert r.get('key2') == ['~']
+assert r.get('key3') == [b'\x80\x01'] # not a printable string so the value read back is bytes
+```
+
+If you want to store integers in your mtbl use varint_encode() and varint_decode() to do so:
```
[the remark this answers is not in the document]
Comment (kelvinatorr) [reply README.md]: Yep; I added that to the readme

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hstern) [reply README.md:91]: Is `r.get('key0')[0]` not taking the first byte of the value of `r.get('key0')` which would truncate a larger varint?

File: mtbl.pyx (hunk 1 above)
Comment (kelvinatorr) [reply mtbl.pyx]: Okay I added a "return_bytes" option when creating a reader, merger, or sorter. The default is False so if you pass in 'abc' you will get back 'abc' like you did in Python2. But if you initialize your reader `return_bytes=True` you'll get back b'abc'

I added a bunch of tests for that functionality as well.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kelvinatorr) [reply README.md:91]: No, reader.get() actually returns a list. So in the 128 example `r.get('key1')` returns `[b'\x80\x01']`. The [0] is just to access the first element for the comparison.

**Review by hstern — APPROVED** [review]

**Pull request merged**